assets/activation-scripts/activate.d/flox-env-helper.awk

```diff
@@ -0,0 +1,194 @@
+#!@nawk@/bin/nawk -f
+#
+# flox-env-helper: grooms and emits commands to set Flox env variables
+#
+# This script inspects environment variables such as PATH and MANPATH and
+# grooms them to put directories implied by the FLOX_ENV_DIRS variable
+# at the front of the list. It also emits commands to set variables in each
+# of the supported [4] shell dialects, based on the value of the FLOX_SHELL
+# environment variable.
+#
+# Usage:
+#  source <(flox-env-helper <shell>) # bash, zsh
+#  flox-env-helper <shell> | source  # fish
+#  eval "`flox-env-helper <shell>`"  # tcsh
+#
+# Note that this script does not expect any input, and the entirety of its
+# logic appears in the BEGIN block.
+#
+# It was formatted using `gawk --pretty-print` and then hand-edited to change
+# tabs back to two spaces and to add back newlines that it had removed from
+# within functions.
+
+BEGIN {
+  # Parse the command line arguments.
+  if (ARGC != 2) {
+    print("Usage: awk -f set-flox-env-vars.awk <shell>") > "/dev/stderr"
+    exit 1
+  }
+  flox_shell = ARGV[1]
+  _flox_shell = basename(flox_shell)
+
+  tracer("BEGIN")
+
+  # Set the default value of FLOX_ENV_DIRS if it is not set.
+  # First check to see if FLOX_ENV is already in FLOX_ENV_DIRS,
+  # and if so then don't change the nesting order. This is particularly
+  # important for the default environment which is reactivated in all
+  # subshells.
+  new_flox_env_dirs = prepend_if_not_found(ENVIRON["FLOX_ENV"], ENVIRON["FLOX_ENV_DIRS"])
+
+  # The FLOX_ENV_LIB_DIRS variable is a direct translation of the FLOX_ENV_DIRS
+  # variable but with each directory having "/lib" appended to it.
+  new_flox_env_lib_dirs = append_dirs("/lib", new_flox_env_dirs)
+
+  # Calculate the values to be prepended to the PATH and MANPATH variables.
+  # First add directories found in FLOX_ENV_DIRS, then add the current value
+  # of FLOX_ENV. Don't worry about duplicating the FLOX_ENV directory as
+  # remove_path_dups will remove dups later.
+  _prepend_path = ""
+  _prepend_manpath = ""
+  for (i = 1; i <= split(new_flox_env_dirs, flox_env_dirs, ":"); i++) {
+    _prepend_path = _prepend_path flox_env_dirs[i] "/bin:" flox_env_dirs[i] "/sbin:"
+    _prepend_manpath = _prepend_manpath flox_env_dirs[i] "/share/man:"
+  }
+  _prepend_path = _prepend_path ENVIRON["FLOX_ENV"] "/bin:" ENVIRON["FLOX_ENV"] "/sbin:"
```

Comment:
Can we document the env variables this depends on in the docs above?
It seems if this script is run without FLOX_ENVS it won't work as expected (or at all?).

Currently that's always the case in practice but it's a weak dependency that we should at least acknowledge

Reply:
Yes I think a check would be good to make that an explicit invariant

Reply:
Added a check in commit 1971aea2fc110340397f1dcca64c9fc7acd28ca4 ... resolving conversation.